R/mockForestPlotData.R

```diff
@@ -12,20 +12,29 @@
 #' mockForestPlotData()
 #' }
 #'
-mockForestPlotData <- function(seed = 1, numRows = 10) {
+mockForestPlotData <- function(seed = 1, numVars = 10) {
   set.seed(seed = seed)
-  res <- dplyr::tibble(
-    group_name = paste0("Outcome ", 1:numRows),
+  for (i in 1:numVars){
+  row1 <- dplyr::tibble(
+    group_name = "Cohort_name",
     group_level = "Cohort 1",
     strata_name = "Overall",
     strata_level = "Overall",
-    variable = paste0("Variable ", 1:numRows),
+    variable = paste0("Variable ", i),
     variable_level = NA,
     variable_type = "numeric",
-    estimate = round((stats::runif(n = numRows, min = 0.51, max = 3.01)),2),
-    estimate_type = "95% confidence interval"
+    estimate = round((stats::runif(n = 1, min = 0.51, max = 3.01)),2),
+    estimate_type = "Central value"
   )
-  res <- res |> dplyr::mutate(estimate = paste0("(",estimate,", ",(estimate - round((stats::runif(n = numRows, min = 0.25, max = 0.75)),2)),", ",(estimate + round((stats::runif(n = numRows, min = 0.25, max = 0.75)),2)),")"))
-  res <- res |> dplyr::mutate(estimate_type = "(Mean, 95% Lower CI, 95% Upper CI)")
-  return(res)
+  row2 <- row1 |> dplyr::mutate(estimate = estimate - round((stats::runif(n = 1, min = 0.25, max = 0.75)),2))
+  row2 <- row2 |> dplyr::mutate(estimate_type = "Lower 95% CI")
+  row3 <- row1 |> dplyr::mutate(estimate = estimate + round((stats::runif(n = 1, min = 0.25, max = 0.75)),2))
+  row3 <- row3 |> dplyr::mutate(estimate_type = "Higher 95% CI")
+
+  if (i==1){
+    result <- rbind(row1,row2,row3)}
+  else{
+    result <- rbind(result,row1,row2,row3)}
+  }
+  return(result)
```

Comment:
Also with the current version the hazard ratios can be < 0, can be add something realistic? we can maybe symmetric confidence intervals from the exponent ( hr = exp(b +- 1.96 * se)